app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -156,6 +156,14 @@ def _validate_feature_configs(self, data):
 
         return data
 
+    def create(self, validated_data):
+        experiment = super().create(validated_data)
+        experiment.reference_branch = NimbusBranch.objects.create(
+            experiment=experiment, slug="control", name="Control", feature_enabled=False
```

Comment:
> Adds an empty control branch one experiment creation (so the control is never null).

Why do we need to do this?

Assuming we leave this in, we need to:
* Update [the logic](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.tsx#L105) around showing both branches initially and only showing one if the user only saved one, which might be tricky, and would otherwise re-open #4899
* Remove `NO_BRANCHES_COPY` and logic around it
* I think we can [take "control" out of `annotateExperimentBranch`](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/reducer/state.ts#L70-L73)
* `FormBranches` Storybook "empty branches" is no longer a valid state, and makes me wonder if we should search for other mock reference branches and fill in the slug/name/feature config to match

Reply:
Basically, the goal of this is to surface validation feedback on the individual fields that make up a branch, and in order to do that the branch needs to first exist -- otherwise the only validation that can occur is on `reference_branch` itself, which would just be null.

There are likely alternatives here, such as looking for the null reference branch validation message in the client and replacing it with an object containing the individual field messages, or finding some way to intercept the null reference branch validation message in the serializer and replace it with a dict containing the individual field messages. But this appears to be the cleanest way. Definitely open to other options that can accomplish this

Reply:
> ...  otherwise the only validation that can occur is on `reference_branch` itself, which would just be null.

Huh, I didn't dig but I wonder why the entire branch comes back `null` since if you create a new experiment, a `referenceBranch` object is created with default values. Sure the branch isn't valid, but it's also still invalid without a branch description filled out.

> ... finding some way to intercept the null reference branch validation message in the serializer and replace it with a dict containing the individual field messages

I don't think this is a bad option, I like it better than doing this in the FE at least. 🤔

I'm OK with the approach if it's the cleanest way but I do think my points in my initial comment have to be addressed. 😩 (or at least the first point, others could be handled in maintenance follow ups, but bleh)

Reply:
> I guess I'm wondering if the branch is actually ever null? Seems like at experiment creation time we get this guy back on the client?

That's coming from [this field resolver](https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/api/v5/types.py#L180:L183). On field lookup if there is a reference branch return it, other instantiate a new one. But this only occurs on field lookup. Actually, if we do keep this change of creating a reference branch on experiment creation I should should remove this bit here as it will be redundant.

> I don't think this is a terrible option, I like it better than doing this in the FE at least.

Yeah this was my initial approach but we thought it might be cleaner to just do the initial branch creation. So I'm a little torn!

@jaredlockhart what do you make of this?

Reply:
So actually, is it possible to just create the branch with `slug` and `name` of the branch set to the defaults, which are empty strings? I think that would actually fix everything. 😆

Reply:
Following up here - seems like we can get away without setting the `slug`, which is great since we can change the `savedBranches`/other similar references in the FE to check branch `slug` instead of `name`, and it doesn't mess with that first bullet point I laid out here.

Reply:
Yeah I think it can go either way, since leaving the description blank forces the user to save which then sets the slug.  Probably fine like this.